config/initializers/004-message_bus.rb

```diff
@@ -119,8 +119,16 @@ def setup_message_bus_env(env)
 end
 MessageBus.reliable_pub_sub.max_backlog_size = GlobalSetting.message_bus_max_backlog_size
 
-MessageBus.long_polling_enabled = SiteSetting.enable_long_polling
-MessageBus.long_polling_interval = SiteSetting.long_polling_interval
+if SiteSetting.table_exists? && SiteSetting.where(name: ['enable_long_polling', 'long_polling_interval']).exists?
+  Discourse.deprecate("enable_long_polling/long_polling_interval have switched from site settings to global settings", drop_from: '2.9.0')
+
+  MessageBus.long_polling_enabled = SiteSetting.enable_long_polling
+  MessageBus.long_polling_interval = SiteSetting.long_polling_interval
+else
+  MessageBus.long_polling_enabled = GlobalSetting.enable_long_polling || true
```

Comment:
If `GlobalSetting.enable_long_polling` is false, then this statement will evaluate to `true`

I think we need a `nil?` check rather than a `truthy` check?

```suggestion
  MessageBus.long_polling_enabled = GlobalSetting.enable_long_polling.nil? ? true : GlobalSetting.enable_long_polling
```

Reply:
Right 🤦